### PR TITLE
feat(generators): add module options

### PR DESCRIPTION
**Pull request by mhartington**
Adds `module` and `no-module` flags to the CLI

Requires https://github.com/ionic-team/ionic-app-scripts/pull/1141 to be merged first

### CODE REVIEW

File: packages/cli-plugin-ionic-angular/src/generate.ts [hunk 1]
```diff
@@ -37,21 +37,25 @@ export async function generate(args: CommandHookArgs): Promise<string[]> {
...
   const [ type, name ] = args.inputs;
+  const includeNgModule = args.options.module;
+  const includePageConstants = args.options.constants;
+
+  // args.env.log.msg(() => `${includePageConstants}`);
 
   switch (type) {
     case 'page':
-      await AppScripts.processPageRequest(context, name);
+      await AppScripts.processPageRequest(context, name, includeNgModule, includePageConstants);
```
Comment (dwieeb) [inline packages/cli-plugin-ionic-angular/src/generate.ts]: Maybe move the options into an object? I can imagine there being a _lot_ more than 2 options for generators. We probably don't want 10+ positional args, especially cross-library.

File: packages/ionic/src/commands/generate.ts [hunk 2]
```diff
@@ -35,6 +35,20 @@ The given ${chalk.green('name')} is normalized into an appropriate naming conven
       description: 'The name of the component being generated',
     }
   ],
+  options: [
```
Comment (dwieeb) [inline packages/ionic/src/commands/generate.ts]: You can provide examples of these options using the `exampleCommands` list above.

File: packages/cli-plugin-ionic-angular/src/generate.ts (hunk 1 above)
Comment (mhartington) [inline packages/cli-plugin-ionic-angular/src/generate.ts]: Makes sense

File: packages/ionic/src/commands/generate.ts (hunk 2 above)
Comment (mhartington) [inline packages/ionic/src/commands/generate.ts]: it is done

**Pull request merged**